internal/mode/static/state/graph/policies_test.go

```diff
@@ -0,0 +1,882 @@
+				" condition is added",
+			policies: map[PolicyKey]*Policy{
+				createPolicyKey(orangeGVK, "orange1"): {
+					Source:    createPolicy(orangeGVK, hrRef, "orange1"),
+					TargetRef: hrTargetRef,
+					Valid:     true,
+				},
+				createPolicyKey(orangeGVK, "orange2"): {
+					Source:    createPolicy(orangeGVK, hrRef, "orange2"),
+					TargetRef: hrTargetRef,
+					Valid:     true,
+				},
+				createPolicyKey(orangeGVK, "orange3-conflicts-with-1"): {
+					Source:    createPolicy(orangeGVK, hrRef, "orange3-conflicts-with-1"),
+					TargetRef: hrTargetRef,
+					Valid:     true,
+				},
+				createPolicyKey(orangeGVK, "orange4"): {
+					Source:    createPolicy(orangeGVK, hrRef, "orange4"),
+					TargetRef: hrTargetRef,
+					Valid:     true,
+				},
+				createPolicyKey(orangeGVK, "orange5-conflicts-with-4"): {
+					Source:    createPolicy(orangeGVK, hrRef, "orange5-conflicts-with-4"),
+					TargetRef: hrTargetRef,
+					Valid:     true,
+				},
+			},
+			fakeValidator: &policiesfakes.FakeValidator{
+				ConflictsStub: func(policy policies.Policy, policy2 policies.Policy) bool {
+					pol1Name := policy.GetName()
+					pol2Name := policy2.GetName()
+
+					if pol1Name == "orange1" && pol2Name == "orange3-conflicts-with-1" {
+						return true
+					}
+
+					if pol1Name == "orange4" && pol2Name == "orange5-conflicts-with-4" {
+						return true
+					}
+
+					return false
+				},
+			},
+			conflictedNames:       []string{"orange3-conflicts-with-1", "orange5-conflicts-with-4"},
+			expConflictToBeCalled: true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			g := NewWithT(t)
+
+			markConflictedPolicies(test.policies, test.fakeValidator)
+
+			if !test.expConflictToBeCalled {
+				g.Expect(test.fakeValidator.ConflictsCallCount()).To(BeZero())
+			} else {
+				g.Expect(test.fakeValidator.ConflictsCallCount()).To(Not(BeZero()))
+				expConflictCond := staticConds.NewPolicyConflicted("Conflicts with another OrangePolicy")
+
+				for key, policy := range test.policies {
+					if slices.Contains(test.conflictedNames, key.NsName.Name) {
+						g.Expect(policy.Valid).To(BeFalse())
+						g.Expect(policy.Conditions).To(ConsistOf(expConflictCond))
+					} else {
+						g.Expect(policy.Valid).To(BeTrue())
+						g.Expect(policy.Conditions).To(BeEmpty())
+					}
+				}
+			}
+		})
+	}
+}
+
+func TestRouteKeyForKind(t *testing.T) {
+	nsname := types.NamespacedName{Namespace: testNs, Name: "route"}
+
+	g := NewWithT(t)
+
+	key := routeKeyForKind(kinds.HTTPRoute, nsname)
+	g.Expect(key).To(Equal(RouteKey{RouteType: RouteTypeHTTP, NamespacedName: nsname}))
+
+	key = routeKeyForKind(kinds.GRPCRoute, nsname)
+	g.Expect(key).To(Equal(RouteKey{RouteType: RouteTypeGRPC, NamespacedName: nsname}))
+
+	rk := func() {
+		_ = routeKeyForKind(kinds.Gateway, nsname)
+	}
+
+	g.Expect(rk).To(Panic())
+}
+
+func createPolicyWithAncestors(numAncestors int) policies.Policy {
```

Comment:
Since this test files lives in the `graph` package, these helper functions will all be exported to the package. Do you think there will be any confusion with that?

Reply:
Yeah, that could be confusing. We do have similar test functions in other test suites. 

What if I added `test` to the names to help disambiguate?

Reply:
Works for me